src/fhost_fpga.py

```diff
@@ -632,13 +689,27 @@ def qdr_okay(self):
 
     def check_fft_overflow(self, wait_time=2e-3):
         """
-        Checks if pfb counters on f-eng are not incrementing i.e. fft is not overflowing
-        :return: True/False
-        """
-        for cnt in range(0, 1):
-            overflow0 = self.registers.pfb_ctrs.read()['data']['pfb_of%d_cnt' % cnt]
-            time.sleep(wait_time)
-            overflow1 = self.registers.pfb_ctrs.read()['data']['pfb_of%d_cnt' % cnt]
+        Checks if pfb counters on f-eng are not incrementing
+            i.e. fft is not overflowing
+        :param: wait_time : Float or None : If not None, fetch overflow counter,
+            wait this long, and fetch a second value; Else, use last read value
+            from cache. Value is not cached if wait_time is None.
+        :return: bool : True if not overflowing
+        """
+        for cnt in range(0, self.num_fengines):
+            if wait_time:
```

Comment:
perhaps, if wait_time is not None.

Difference is that someone might pass a wait_time of 0 if they want to do the test as quickly as possible without caching values.